### DISCUSSION TITLE
[Question][Hw4] Usage and meaning of order

### DISCUSSION THREAD

**Discussion by qawbecrdtey** [prosyslab-classroom/cs524-program-analysis#375] [Q&A]
In each modules `Sign` and `Interval`, there are functions `order` and `cmp`. I believe `cmp` is for comparison between integers and `order` is for ordering between abstract elements. However, I couldn't figure out what the output should be if two elements are not in a single chain.

**Comment by KihongHeo** [comment]
It is obviously the same as the following definition: if the two elements are ordered, then true. Otherwise false.

**Comment by qawbecrdtey** [comment]
So does it mean `order x y` is true if and only if they are on a same chain, regardless of their order `x <= y` or `x >= y`?

**Comment by KihongHeo** [comment]
`order x y` iff $x \sqsubseteq y$.

**Comment by qawbecrdtey** [comment]
Thank you for the reply!
